runtime/metricsview/executor/executor_query_attributes.go

```diff
@@ -0,0 +1,47 @@
+package executor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/parser"
+)
+
+// resolveQueryAttributes resolves the query attributes defined in metrics views
+func (e *Executor) resolveQueryAttributes(ctx context.Context) (map[string]string, error) {
+	if len(e.metricsView.QueryAttributes) == 0 {
+		return nil, nil
+	}
+
+	claims := runtime.ClaimsFromContext(ctx)
```

Comment:
Attaching claims to the `ctx` is only something we do in places where there are no other option, such as in the server. It is not safe to rely on this in internal packages because you don't know what upstream logic may call this code, and whether it added stuff to the context or not. 

So values like this should be passed in explicitly through `executor.New`.